packages/react-compose/src/next/simplifyShorthand.tsx

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+import { GenericDictionary } from './types';
+
+/**
+ * Ensures that the given slots are represented using object syntax. This ensures that
+ * the object can be merged along with other objects.
+ * @param props - The incoming props
+ * @param shorthandPropNames - An array of prop names to apply simplification to
+ */
+export const simplifyShorthand = (props: GenericDictionary, shorthandPropNames?: string[]) => {
+  let newProps = props;
+
+  if (shorthandPropNames && shorthandPropNames.length) {
+    newProps = {
+      ...props,
+    };
+    for (const propName of shorthandPropNames) {
+      const propValue = props[propName];
+
+      if (propValue !== undefined && (typeof propValue !== 'object' || React.isValidElement(propValue))) {
+        newProps[propName] = { children: propValue };
+      }
+    }
+  }
+
+  return newProps;
+};
```

Comment:
```jsx
console.log(simplifyShorthand({ image: false }, ["image"]));
console.log(simplifyShorthand({ image: true }, ["image"]));
```

![image](https://user-images.githubusercontent.com/14183168/89167067-3c71b500-d57b-11ea-8f8e-29367c0e4cfe.png)

Is it expected?

Reply:
yep. It's analogous to: `{ label: 'value' }, ['label']` returning `{ label: { children: 'value' }}`. Render the value as children of the slot.

The `mappedProp` slot feature is unaccounted for here. I could add it as needed, just not sure on the best solution other than to change the argument to an options bag and then have something like:

```
{
  props,
  shorthandPropNames,
  propMappings: {
    image: 'src'
  }
}
```

Reply:
If it should do something different, we can certainly tweak.